### PR TITLE
Upgrade React and Chakra UI

### PR DESCRIPTION
**Pull request by Belco90**
## Changes

- Upgrade React to v18
- Upgrade Chakra UI to v2, replacing deprecated features usages
- Fix nested HTML validation (`div` cannot appear as a descendant of `p`)

## Context

Chakra was the last dependency pending to be compatible with React v18, so I'm upgrading both at the same time since Chakra v2 is only compatible with React v18.

## Tests

I've checked my work by:

- [ ] Adding new tests or adjusting existing tests
- [X] Testing manually

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -74,55 +74,55 @@ const FEATURES_DESCRIPTIONS: Array<FeatureItemProps> = [
     icon: HiOutlineSwitchHorizontal,
     title: 'Compare releases easily',
     children: (
-      <Box>
+      <Text as="span">
```
Comment (Belco90) [inline src/pages/index.tsx:77]: This change isn't fine: it's modifying the UI. I'll address it properly.

Comment (Belco90) [reply src/pages/index.tsx]: Should be fixed now

**Comment by Belco90** [comment]
It turns out the Happo screenshots were incorrect before, and after this upgrade they are correct!

Setting the PR as draft tho, since one of the Chakra subdependencies is not 100% compatible with React v18 yet (they'll release it soon).

**Comment by HonkingGoose** [comment]
> It turns out the Happo screenshots were incorrect before, and after this upgrade they are correct!

Yay! 🥳 

> Setting the PR as draft tho, since one of the Chakra subdependencies is not 100% compatible with React v18 yet (they'll release it soon).

In that case it makes sense to wait to have everything compatible with React v18 and be done with it at once. 😄

**Comment by Belco90** [comment]
Chakra UI upgraded to the latest version, which is fully compatible with React v18 🤞

**Comment by Belco90** [comment]
@HonkingGoose this is ready to go!

**Review by HonkingGoose — APPROVED** [review]
Looks good to me.

**Pull request merged**